sources/iTermColorMap.m

```diff
@@ -113,7 +125,7 @@ - (NSColor *)colorForKey:(iTermColorMapKey)theKey {
         return [NSColor colorWith8BitRed:red green:green blue:blue];
     } else {
         NSColor *result = _map[@(theKey)];
-        if (!result) {
+        if (!result && theKey != kColorMapUnderline) {
```

Comment:
It's ok to just return result here. The code to return red was to catch bugs, but it hasn't been needed any more.